include/big_core_base.h

```diff
@@ -116,7 +120,7 @@ class BigCoreBase
     uint64_t dataLength = 0;
     uint64_t memorySize = DEFAULT_MEMORY_SIZE;
 
-    std::vector<uint64_t> outermostEntitiesOffsets;
+	std::vector<uint64_t> outermostEntitiesOffsets = { 0 };
```

Comment:
Tady stale nechapu, proc tam davas ten jeden prvek s hodnotou nula a nenechas prazdny vektor dokud nebudeme vedet neco o vstupnich datech.

Reply:
Je to myslene tak, ze kazdy dalsi prvek bude ukazovat za posledni data
jako treba tady u CSR maice: https://en.wikipedia.org/wiki/Sparse_matrix

Reply:
Tady si opet asi nerozumime kvuli tomu, na co se ptas nize. Objekt typu BigCoreInput (tj. instance tridy BigCoreInput) bude nacitat pouze jeden soubor! Pokazde, kdyz bude uzivatel chtit ziskat nejaka data, vytvori si objekt typu BigCoreInput, nastavi jmeno souboru a pak uz muze cist libovolne polozky. Jakmile chce ale cist jiny soubor, musi si vytvorit novy objekt (nebo ve stavajicim objektu zavrit soubor a otevrit novy, ale tim se vsechna stara data zahodi a vyjde to uplne nastejno, jako kdyby si vytvoril ten novy objekt).

Tato promenna outermostEntitiesOffsets slouzi pro interni ucely a pri otevreni souboru je potreba nastavit jeji hodnoty tak, aby rikaly, kde v datech zacina ktery obrazek. Priklad: pro jednoduchost numberOfTiles = 1, numberOfImages = 5, width * height * numberOfPlanes = 1000, dataType = 5 (unsigned char 8-bit). Pak outermostEntitiesOffsets = {0, 1000, 2000, 3000, 4000}. Pokud chci napriklad data pro obrazek s indexem 2 (treti obrazek), musim k pointeru na zacatek dat pricist 2000. Jiny priklad: vse jako predtim, ale dataType = {2, 2, 3, 2, 2} (2 = float 32-bit, 3 = double 64-bit). Pak outermostEntitiesOffsets = {0, 4000, 8000, 16000, 20000}. Je zbytecne si pamatovat offset za posledni data (24000).

Reply:
Aha, ted uz to cele chapu.

Reply:
nicmene mit ukazatel za data se hodi. Pak muzu napocitat velikost obrazku primo z toho vektoru a nebudu muset volat metodu getImageSize()